tezos/encoding/src/binary_writer.rs

```diff
@@ -367,6 +407,31 @@ fn encode_value(data: &mut Vec<u8>, value: &Value, encoding: &Encoding) -> Resul
                     Error::custom("Encoded message size overflow while encoding a dynamic value")
                 })
         }
+        Encoding::BoundedDynamic(max, dynamic_encoding) => {
+            let data_len_before_write = data.len();
+            // put 0 as a placeholder
+            data.put_u32(0);
+            // we will use this info to create slice of buffer where inner record size will be stored
+            let data_len_after_size_placeholder = data.len();
+
+            // write data
+            let bytes_sz = encode_value(data, value, dynamic_encoding)?;
+            if bytes_sz > *max {
+                return Err(bound_error(encoding, *max, bytes_sz));
+            }
+
+            // capture slice of buffer where List length was stored
+            let mut bytes_sz_slice =
+                &mut data[data_len_before_write..data_len_after_size_placeholder];
+            // update size
+            bytes_sz_slice.write_u32::<BigEndian>(bytes_sz as u32)?;
```

Comment:
Is there a better way to perform this cast `bytes_sz as u32` (from `usize` to `u32`) ?

Reply:
Sure.